theme/additional.js

```diff
@@ -36,3 +36,33 @@
     }
   });
 })();
+
+function setCodeToPlayground() {
+  var codes = JSON.parse(localStorage.getItem(window.location.href));
+  if (codes) {
+    var i = 0;
+    Array.from(document.querySelectorAll(".playground")).forEach(function (
+      pre_block
+    ) {
+      let code_block = pre_block.querySelector("code");
+      let editor = window.ace.edit(code_block);
+      editor.setValue(codes[i]);
+      editor.clearSelection();
+      i += 1;
+    });
+  }
+}
+setCodeToPlayground();
+function getCodeFromPlayground() {
+  var codes = [];
+  Array.from(document.querySelectorAll(".playground")).forEach(function (
+    pre_block
+  ) {
+    let code_block = pre_block.querySelector("code");
+    let editor = window.ace.edit(code_block);
+    let code = editor.getValue();
+    codes.push(code);
+  });
+  localStorage.setItem(window.location.href, JSON.stringify(codes));
+}
+setInterval(getCodeFromPlayground, 900);
```

Comment:
This seems like it will wake up the browser frequently even if nothing has changed. I wonder if we could either do this on some event indicating the browser is navigating away (maybe https://developer.mozilla.org/en-US/docs/Web/API/Window/pagehide_event?) or on a timer after the user has modified something in the editor.